planner/core/rule_partition_processor.go

```diff
@@ -191,8 +191,16 @@ func (s *partitionProcessor) findUsedPartitions(ctx sessionctx.Context, tbl tabl
 						used = append(used, int(idx))
 					}
 				} else {
-					used = []int{FullRange}
-					break
+					maxUsedPartitions := 1 << col.RetType.Flen
```

Comment:
`Flen` may be very large or equal to -1, please check to ensure that no corner case appears.

Reply:
OK

Reply:
I think that as for `ETInt` type partition key, the `Flen` cannot be  -1 which is used as `UnspecifiedLength`. As for the fact that `Flen` may be very large, the code now has been modified to prevent overflow. PTAL

Reply:
For safety, I suggest that we check `Flen != -1` also here since in my experience this assumption is easy to break...
You can see that there are lots of `TODO` about `Flen` in `typeinfer_test.go`, which can indicate that we don't maintain `Flen` very carefully.
![image](https://user-images.githubusercontent.com/7499936/117235971-53bb3700-ae5a-11eb-83bf-f68acad48e90.png)